[BUG] Test failing on CMake 4.0.2 due to no minimum version set

**Describe the bug**
On CMake 4.0.2-dirty, `tests/docs/bare_bones_cmake` fails due to no `cmake_minimum_required()` command being present. CMake 4 requires a minimum version now (see [CMP0000](https://cmake.org/cmake/help/latest/policy/CMP0000.html)).

**To Reproduce**
Steps to reproduce the behavior:
1. Make sure you are using CMake >= 4.0.
2. Build CMaize
```
cmake -H. -Bbuild -DBUILD_TESTING=ON
```
4. Run the unit tests and you will get a warning about no minimum version set:
```
cd build
ctest --output-on-failure -j "$(nproc)" -LE integration_tests
```
5. Error that occurs:
```
CMake Error in CMakeLists.txt:
  No cmake_minimum_required command is present.  A line of code such as

    cmake_minimum_required(VERSION 4.0)

  should be added at the top of the file.  The version specified may be lower
  if you wish to support older CMake versions for this project.  For more
  information run "cmake --help-policy CMP0000".


-- Configuring incomplete, errors occurred!
CMake Error at /home/zachcran/programs/cmakepp_workspace/cmaize/tests/conf_build_test.cmake:105 (execute_process):
  execute_process failed command indexes:

    1: "Child return code: 1"
```

**Additional context**
I also tested this with CMake 3.26.0-rc4 and the error did not occur.

--- a/docs/src/background/redundancy.rst
+++ b/docs/src/background/redundancy.rst
@@ -60,7 +60,7 @@ then the minimal ``CMakeLists.txt`` file looks like:
 
 .. literalinclude:: /../../tests/docs/bare_bones_cmake/CMakeLists.txt
    :language: CMake
-   :lines: 15
+   :lines: 18-20
    :linenos:
    :lineno-start: 1
 

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -13,5 +13,10 @@ include(cmake_test/cmake_test)
 
 set(BUILD_TESTING "${build_testing_old}" CACHE BOOL "" FORCE)
 
-ct_add_dir("cmaize" USE_REL_PATH_NAMES)
+ct_add_dir(
+    "cmaize"
+    USE_REL_PATH_NAMES
+    CMAKE_OPTIONS
+        -DCMAKE_MESSAGE_LOG_LEVEL=VERBOSE
+)
 add_subdirectory(docs)

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -13,6 +13,5 @@ include(cmake_test/cmake_test)
 
 set(BUILD_TESTING "${build_testing_old}" CACHE BOOL "" FORCE)
 
-# ct_add_dir("cpp")
-ct_add_dir("cmaize")
+ct_add_dir("cmaize" USE_REL_PATH_NAMES)
 add_subdirectory(docs)

--- a/tests/docs/bare_bones_cmake/CMakeLists.txt
+++ b/tests/docs/bare_bones_cmake/CMakeLists.txt
@@ -15,7 +15,7 @@
 # Minimum version required as of CMake 4.0
 # (see https://cmake.org/cmake/help/latest/policy/CMP0000.html).
 # Match this with minimum version from 'CMaize/CMakeLists.txt'.
-cmake_minimum_required(VERSION 3.19)
+cmake_minimum_required(VERSION 3.5)  # Required as of CMake 4.0
 
 add_executable(hello_world hello_world.cpp)
 

--- a/tests/docs/bare_bones_cmake/CMakeLists.txt
+++ b/tests/docs/bare_bones_cmake/CMakeLists.txt
@@ -12,6 +12,11 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Minimum version required as of CMake 4.0
+# (see https://cmake.org/cmake/help/latest/policy/CMP0000.html).
+# Match this with minimum version from 'CMaize/CMakeLists.txt'.
+cmake_minimum_required(VERSION 3.19)
+
 add_executable(hello_world hello_world.cpp)
 
 # The below lines of code are used in unit testing this code example

